### PR TITLE
poetry

### PR DESCRIPTION
**Pull request by vaaaaanquish**
@hirosassa @Hi-king 
```
$ poetry build
Building gokart (0.3.24.post1.dev0+ca04202)
  - Building sdist
  - Built gokart-0.3.24.post1.dev0+ca04202.tar.gz
  - Building wheel
  - Built gokart-0.3.24.post1.dev0+ca04202-py3-none-any.whl

$ pip install dist/gokart-0.3.24.post1.dev0+ca04202-py3-none-any.whl
...
Successfully installed gokart-0.3.24.post1.dev0+ca04202
```

### CODE REVIEW

**Comment by vaaaaanquish** [comment]
# TODO
- [ ] set secrets env
- [ ] Disconnect the TravisCI

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -14,14 +14,14 @@ jobs:
...
-      uses: pypa/gh-action-pypi-publish@master
-      with:
-        user: __token__
-        password: ${{ secrets.PYPI_API_TOKEN }}
+        python -m pip install --upgrade pip
+        pip install poetry poetry-dynamic-versioning twine
+    - name: Build and publish
+      env:
+        TWINE_USERNAME: ${{ secrets.PYPI_USERNAME }}
+        TWINE_PASSWORD: ${{ secrets.PYPI_PASSWORD }}
```
Comment (hirosassa) [inline .github/workflows/publish.yml:25]: We can use TWINE_USERNAME as `__token__` like below
https://github.com/m3dev/gokart/blob/master/.github/workflows/publish.yml#L26-L27

ref: 
https://pypi.org/help/#apitoken

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+[tool.poetry-dynamic-versioning]
+enable = true
+style = "pep440"
+pattern = "^(?P<base>\\d+\\.\\d+\\.\\d+)"
+
+[tool.poetry.dependencies]
+python = "^3.6"
+luigi = "*"
+boto3 = "*"
+slack_sdk = "*"
```
Comment (hirosassa) [inline pyproject.toml:21]: for slack api pagination feature, we need below

`slack-sdk = "^3.0"`

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+pandas = "*"
```
Comment (hirosassa) [inline pyproject.toml:22]: for unit testing, we need below
`pandas = "<1.2"`

**Review by hirosassa — COMMENTED** [review]
Thanks for the PR! comments added

File: pyproject.toml (hunk 3 above)
Comment (vaaaaanquish) [reply pyproject.toml:22]: @hirosassa Thanks
ref: https://github.com/pandas-dev/pandas/issues/39247

File: pyproject.toml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+repository = "https://github.com/m3dev/gokart"
+documentation = ""
+
+[tool.poetry-dynamic-versioning]
+enable = true
+style = "pep440"
+pattern = "^(?P<base>\\d+\\.\\d+\\.\\d+)"
+
+[tool.poetry.dependencies]
+python = "^3.7"
```
Comment (Hi-king) [inline pyproject.toml:18]: [IMO] Too strict? 3.6 might be OK?

Comment (vaaaaanquish) [reply pyproject.toml:18]: Because pandas is not supported python 3.6.
https://pypi.org/project/pandas/

Comment (Hi-king) [reply pyproject.toml:18]: Wow. Thx for that info!

**Review by Hi-king — APPROVED** [review]
LGTM

**Review by hirosassa — APPROVED** [review]
LGTM! Thanks

**Comment by vaaaaanquish** [comment]
gogogo!

**Pull request merged**